File: NetFramework/FlexGrid/CS/RowDetails/Samples/InheritanceSample.cs
using System;
using System.Data;
using System.Windows.Forms;
using C1.Win.C1FlexGrid;

namespace RowDetails.Samples
{
    public partial class InheritanceSample : UserControl
    {
        public InheritanceSample()
        {
            InitializeComponent();
        }

        private void InheritanceSample_Load(object sender, EventArgs e)
        {
            if (DesignMode) return;

            string conn = Util.GetConnectionString();
            var ds = new DataSet();
            Util.FillTable(ds, "Employees", conn);

            flexGrid.DataSource = ds;
            flexGrid.DataMember = "Employees";
            flexGrid.RowDetailProvider = (g, r) => new DerivedRowDetail();
            flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected;
            flexGrid.Cols["Notes"].Visible = false;
        }
    }
}

File: NetFramework/FlexGrid/VB/RowDetails/Samples/InheritanceSample.vb
﻿Imports C1.Win.C1FlexGrid

Public Class InheritanceSample
    Private Sub InheritanceSample_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim conn As String = Util.GetConnectionString()
        Dim ds = New DataSet()
        Util.FillTable(ds, "Employees", conn)

        flexGrid.DataSource = ds
        flexGrid.DataMember = "Employees"

        flexGrid.RowDetailProvider = Function(g, r) New DerivedRowDetail()
        flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected

        flexGrid.Cols("Notes").Visible = False
    End Sub
End Class
